{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6\n"
     ]
    }
   ],
   "source": [
    "import PyPDF2\n",
    " \n",
    "#create file object variable\n",
    "#opening method will be rb\n",
    "pdffileobj=open('PMC2538882.pdf','rb')\n",
    " \n",
    "#create reader variable that will read the pdffileobj\n",
    "pdfreader=PyPDF2.PdfReader(pdffileobj)\n",
    " \n",
    "#This will store the number of pages of this pdf file\n",
    "x=len(pdfreader.pages)\n",
    " \n",
    "print(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "output = \"\"\n",
    "for i in range(len(pdfreader.pages)):\n",
    "    pageObj = pdfreader.pages[i]\n",
    "    output += pageObj.extract_text()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The T790M mutation in EGFR kinase causes drug\n",
      "resistance by increasing the affinity for ATP\n",
      "Cai-Hong Yun*†, Kristen E. Mengwasser†, Angela V. Toms*†, Michele S. Woo‡, Heidi Greulich‡§, Kwok-Kin Wong‡¶,\n",
      "Matthew Meyerson‡§/H20648, and Michael J. Eck*†**\n",
      "Departments of *Biological Chemistry and Molecular Pharmacology and/H20648Pathology, Harvard Medical School, 25 Shattuck Street, Boston,\n",
      "MA 02115; Departments of†Cancer Biology and‡Medical Oncology, Dana-Farber Cancer Institute, 44 Binney Street, Boston, MA 02115;\n",
      "¶Department of Medicine, Brigham and Women’s Hospital, Boston, MA 02115; and§The Broad Institute of Harvard and Massachusetts\n",
      "Institute of Technology, 320 Charles Street, Cambridge, MA 02141\n",
      "Edited by Harold E. Varmus, Memorial Sloan–Kettering Cancer Center, New York, NY, and approved December 13, 2007 (received for review\n",
      "October 11, 2007)\n",
      "Lung cancers caused by activating mutations in the epidermal\n",
      "growth factor receptor (EGFR) are initially responsive to smallmolecule tyrosine kinase inhibitors (TKIs), but the efﬁcacy of theseagents is often limited because of the emergence of drug resistanceconferred by a second mutation, T790M. Threonine 790 is the‘‘gatekeeper’’ residue, an important determinant of inhibitorspeciﬁcity in the ATP binding pocket. The T790M mutation hasbeen thought to cause resistance by sterically blocking binding ofTKIs such as geﬁtinib and erlotinib, but this explanation is difﬁcultto reconcile with the fact that it remains sensitive to structurallysimilar irreversible inhibitors. Here, we show by using a directbinding assay that T790M mutants retain low-nanomolar afﬁnityfor geﬁtinib. Furthermore, we show that the T790M mutationactivates WT EGFR and that introduction of the T790M mutationincreases the ATP afﬁnity of the oncogenic L858R mutant by morethan an order of magnitude. The increased ATP afﬁnity is theprimary mechanism by which the T790M mutation confers drugresistance. Crystallographic analysis of the T790M mutant showshow it can adapt to accommodate tight binding of diverse inhib-itors, including the irreversible inhibitor HKI-272, and also suggestsa structural mechanism for catalytic activation. We conclude thatthe T790M mutation is a ‘‘generic’’ resistance mutation that willreduce the potency of any ATP-competitive kinase inhibitor andthat irreversible inhibitors overcome this resistance simply throughcovalent binding, not as a result of an alternative binding mode.\n",
      "lung cancer /H20841tyrosine kinase /H20841x-ray crystallography\n",
      "Mutations in the tyrosine kinase domain of the epidermal\n",
      "growth factor receptor (EGFR) have recently been iden-\n",
      "tified as a cause of nonsmall cell lung cancer (1–7). The mostcommon oncogenic mutations are small, in-frame deletions inexon 19 and a point mutation that substitutes Leu-858 witharginine (L858R). These mutations likely cause constitutiveactivation of the kinase by destabilizing the autoinhibited con-\n",
      "formation (8, 9), which is normally maintained in the absence ofligand stimulation. Importantly, the activating mutations havealso been found to confer sensitivity to the small moleculetyrosine kinase inhibitors (TKIs) gefitinib and erlotinib (1–3). Asfirst reported by Carey et al. (10) in studies with erlotinib, the\n",
      "mutant kinases bind the inhibitors more tightly than does the WTEGFR and additionally the deletion and L858R mutationsmarkedly decrease the affinity of the kinase for ATP (8, 10), withwhich the inhibitors compete for binding. These two effectscombine to yield the remarkable potency of gefitinib anderlotinib against tumors and cell lines that are ‘‘addicted’’ to theactivated EGFR for survival (5, 11, 12).\n",
      "Clinically, the efficacy of these TKIs is often of limited\n",
      "duration because of the emergence of drug resistance conferredby a second mutation: substitution of threonine 790 with me-thionine (T790M) (13–15). The T790M mutation accounts forabout half of all resistance to gefitinib and erlotinib (16, 17).Threonine 790 is the gatekeeper residue in EGFR, so namedbecause its key location at the entrance to a hydrophobic pocket\n",
      "in the back of the ATP binding cleft makes it an importantdeterminant of inhibitor specificity in protein kinases. Substitu-tion of this residue in EGFR with a bulky methionine has beenthought to cause resistance by steric interference with binding ofTKIs, including gefitinib and erlotinib (13–15). However, theT790M mutant kinase remains sensitive to irreversible inhibi-tors, including CL-387,785, EKB-569, and HKI-272 (14, 15,18–20). These compounds closely resemble the reversible ani-linoquinazoline inhibitors, but contain a reactive Michael-acceptor group that forms a covalent bond with Cys-797 at theedge of the ATP-binding cleft (Fig. 1). The irreversible inhibitorsare designed to target only this cysteine in EGFR because oftheir specific noncovalent interactions in the ATP bindingpocket, which resemble those of reversible anilinoquinazolinecompounds. Thus the fact that these irreversible TKIs still inhibitthe T790M mutant is at odds with steric hindrance as a mech-anism of resistance: the reversible inhibitor gefitinib and theirreversible inhibitor EKB-569 have identical aniline substitu-ents that are expected to bind in the gatekeeper pocket (Fig. 1),so the same steric effects that block gefitinib binding should alsoprevent the initial binding of EKB-569 (and of the relatedcompound HKI-272).\n",
      "A number of observations indicate that in addition to confer-\n",
      "ring drug resistance, the gatekeeper mutation may derepress the\n",
      "catalytic activity of EGFR and other kinases. A germ-lineT790M mutation has been discovered in a family with a hered-itary predisposition to lung cancer, suggesting that this mutationconfers a growth advantage in the absence of the selectivepressure of TKIs (21). Consistent with this idea, introduction ofthe T790M in tandem with the L858R mutant in NIH 3T3 cellsincreases EGFR activity and enhances the transformed pheno-type (22). Transgenic mice engineered with lung-specific expres-sion of the T790M mutant develop lung adenocarcinomas (23),albeit with a longer latency than those harboring the L858R orcombined L858R and T790M mutations (23, 24). The EGFRT790M mutation was also identified in an untreated case ofBarrett’s esophagus and the corresponding adenocarcinoma(25). Interestingly, the corresponding mutation in BCR-Abl\n",
      "Author contributions: C.-H.Y., K.E.M., A.V.T., M.S.W., H.G., M.M., and M.J.E. designed\n",
      "research; C.-H.Y., K.E.M., A.V.T., M.S.W., and H.G. performed research; K.-K.W. contributednew reagents/analytic tools; C.-H.Y., K.E.M., A.V.T., M.S.W., H.G., K.-K.W., M.M., and M.J.E.analyzed data; and C.-H.Y. and M.J.E. wrote the paper.\n",
      "Conﬂict of interest statement: M.J.E. and M.M. are consultants for and receive research\n",
      "funding from Novartis Institutes for Biomedical Research.\n",
      "This article is a PNAS Direct Submission.Data deposition: The crystallographic coordinates and structure factors have been depos-\n",
      "ited in the Protein Data Bank, www.pdb.org (PDB ID codes 2JIT, 2JIU, and 2JIV).\n",
      "**To whom correspondence should be addressed. E-mail: eck@red.dfci.harvard.edu.This article contains supporting information online at www.pnas.org/cgi/content/full/\n",
      "0709662105/DC1 .\n",
      "© 2008 by The National Academy of Sciences of the USA\n",
      "2070–2075 /H20841PNAS /H20841February 12, 2008 /H20841vol. 105 /H20841no. 6 www.pnas.org /H20862cgi /H20862doi /H2086210.1073 /H20862pnas.0709662105\n",
      "(T315I) confers resistance to imatinib and other TKIs in the\n",
      "treatment of chronic myelogenous leukemia and has also beenfound to preexist in untreated CML (26, 27). The equivalentmutation is found in v-Src (T338I) and has long been known toconfer transforming activity on c-Src (28). Despite the long\n",
      "history of interest in this key residue in control of tyrosine kinaseactivity, a structural understanding of its effects is lacking. Tobetter understand its role in inhibitor resistance and kinase\n",
      "Fig. 1. Chemical structures of selected EGFR inhibitors. All compounds are drawn in a consistent orientation and conformation that reﬂects their approxima te\n",
      "binding mode in the EGFR kinase. HKI-272 and EKB-569 are examples of irreversible inhibitors. Lapatinib and HKI-272 are thought to require the inacti ve\n",
      "conformation of EGFR for binding because of their additional aniline substitutions.\n",
      "Yun et al. PNAS /H20841February 12, 2008 /H20841vol. 105 /H20841no. 6 /H208412071\n",
      "MEDICAL SCIENCES\n",
      "deregulation, we have studied the structural and enzymological\n",
      "effects of the T790M mutation in the context of both the WT andthe L858R-mutant EGFR kinases.\n",
      "Results\n",
      "T790M Mutants Bind Gefitinib with Low Nanomolar Affinity. We first\n",
      "measured binding of gefitinib to the WT, L858R, T790M, andL858R/T790M mutants by using a direct binding assay in whichintrinsic fluorescence of EGFR is quenched by titration withthe inhibitor (8). Strikingly, the T790M mutation only mod-estly affects binding of gefitinib in the context of the L858Rmutant (Table 1). The resistant L858R/T790M double-bindsgefitinib with K\n",
      "d/H1100510.9 nM, which is only /H110154-fold weaker than\n",
      "the exquisitely sensitive L858R mutant ( Kd/H110052.4 nM). The\n",
      "T790M mutant binds gefitinib with Kd/H110054.6 nM, nearly as\n",
      "tightly as the L858R mutant and considerably tighter than theWT kinase. The small difference in gefitinib affinity caused byintroduction of the secondary T790M mutation is in starkcontrast to the roughly two orders of magnitude differencesobserved in the sensitivity of cell lines bearing the L858R vs.L858R/T790M or exon 19 deletions with T790M mutations(13–15, 29), and therefore cannot explain the clinically ob-\n",
      "served drug resistance. We also examined binding of thepyrrolopyrimidine compound AEE788 (Novartis Pharmaceu-ticals), which binds in a manner similar to gefitinib despite thedifference in chemical scaffold (8). The T790M mutant has amore dramatic effect on the affinity for AEE788, but notablythe L858R/T790M double mutant retains 18.6 nM affinity forthis compound (as compared with K\n",
      "d/H110051.1 nM for the L858R\n",
      "mutant). The larger effect on AEE788 as compared withgefitinib is not unexpected because the phenethylamine sub-stituent on this inhibitor extends further into the hydrophobicpocket that is ‘‘guarded’’ by the gatekeeper residue (8).\n",
      "Crystal Structures of T790M Mutant. Crystal structures of the\n",
      "T790M mutant show how inhibitors are accommodated in thepresence of the gatekeeper mutation, in both the active andinactive conformations of the kinase. We determined struc-tures of the T790M mutant alone and in complex with theirreversible inhibitor HKI-272 in the inactive conformation orin complex with AEE788 in the active conformation [seesupporting information (SI) Table 3 for crystallographic sta-\n",
      "tistics]. The structure of the T790M mutant in complex withAEE788 is shown in Fig. 2 A. The compound binds in essen-\n",
      "tially the same manner observed in the WT enzyme, with thepyrrolopyrimidine core making two hydrogen bonds with thehinge region of the kinase and the phenethylamine substituentextending into the gatekeeper hydrophobic pocket. Compar-ison with the binding of AEE788 to the WT enzyme revealsonly a small rotation of the phenethylamine substituent, whichis in direct contact with the mutant gatekeeper residue.Comparison of the AEE788 complex with the structure of theT790M mutant in the absence of inhibitor (Apo-T790M) showsthat the Met-790 side chain must adopt a different rotamer toaccommodate the inhibitor (Fig. 2 B).The irreversible inhibitor HKI-272 is a 4-(arylamino)quino-\n",
      "line-3-carbonitrile compound and a potent inhibitor of bothEGFR and ErbB2 kinases (14, 30). In complex with HKI-272,the EGFR kinase adopts an inactive conformation in which theregulatory C-helix is displaced from its active position (Fig.2C). The enlarged hydrophobic pocket created by the outward\n",
      "rotation of the C-helix appears to be required to accommodatethe bulky aniline substituent found in HKI-272. Both HKI-272and lapatinib contain additional aromatic groups appended tothe aniline ring (a 2-pyridinyl group in HKI-272 and a fluoro-phenyl group in lapatinib; Fig. 1). Thus it is not surprising thatHKI-272, like lapatinib, binds the inactive conformation of thekinase and that the overall binding mode of the two com-pounds is similar (Fig. 2 D). The quinoline core of HKI-272\n",
      "forms a single hydrogen bond with the hinge region of thekinase in a manner analogous to anilinoquinazoline com-pounds (31, 32). The 2-pyridinyl group of HKI-272 is sur-rounded by hydrophobic residues in the expanded pocket,including Met-766 in the C-helix, Phe-856, and Met-790, themutant gatekeeper residue. The nitrile substituent of HKI-272also approaches the gatekeeper residue (extending to /H110153Å\n",
      "from the methionine side chain). In addition to these nonco-valent interactions of HKI-272, the expected covalent bond isformed between Cys-797 at the edge of the active site cleft andthe crotonamide Michael-acceptor group on the inhibitor,rendering binding irreversible (Fig. 2 C). Although the reso-\n",
      "lution of the structure is modest, electron density for theinhibitor and for the covalent bond is clear ( SI Fig. 4 ).\n",
      "The structure of the T790M mutant also suggests a possible\n",
      "mechanism of catalytic activation. We hypothesize that themutation facilitates interconversion between the inactive andactive conformations via direct interaction with the Asp-Phe-Glysequence (DFG motif) at the base of the kinase activation loop(see SI Figs. 5 and 6 and related discussion in SI Text ). The\n",
      "mutation may also enhance the stability of the active confor-mation (relative to the inactive), as it makes favorable hydro-phobic interactions with Met-766 and Leu-777 in the active state.\n",
      "Increased ATP Affinity of the L858R/T790M Mutant Confers Drug\n",
      "Resistance. The binding data and crystal structures clearly dem-\n",
      "onstrate that the gatekeeper mutation does not sterically blockbinding of reversible inhibitors. Why then does the T790Mmutation confer resistance? Kinetic characterization of the WTand mutant EGFR kinases reveals a marked decrease in theMichaelis-Menten constant ( K\n",
      "m) for ATP in the drug-resistant\n",
      "L858R/T790M mutant as compared with the drug-sensitiveL858R mutant (Table 2). As described (8, 10), the L858R mutantactivates EGFR, but also reduces the apparent affinity for ATP(Table 2). Strikingly, the T790M mutation restores the ATPaffinity to near WT levels in the L858R/T790M double mutant(K\n",
      "m[ATP] /H110058.4/H9262M, as compared with Km[ATP] /H11005148/H9262M for the\n",
      "L858R mutant). In isolation, the T790M mutation does notsignificantly affect ATP affinity. We cannot explain structurallywhy the T790M mutation increases ATP affinity in the contextof the L858R mutant, but not in the context of the WT enzyme.\n",
      "We also find that the T790M mutation activates the kinase\n",
      "/H110155-fold as compared with the WT enzyme (Table 2); this\n",
      "catalytic activation of the T790M mutant likely explains itspresence as a germ-line mutation in a family predisposed to lungcancer (21). Although the L858R/T790M mutant has a modestlydecreased k\n",
      "catrelative to the L858R mutant, it is still much more\n",
      "active than the WT enzyme and also exhibits a 5-fold higherk\n",
      "cat/Km[ATP] than the L858R mutant (Table 2).\n",
      "Because TKIs such as gefitinib must compete with ATP for\n",
      "binding to the kinase active site, the enhanced ATP affinity isexpected to decrease the apparent inhibitor potency. In theL858R/T790M mutant this ‘‘ K\n",
      "meffect’’ combines with the small\n",
      "difference in binding affinity for gefitinib to dramatically de-Table 1. Inhibitor dissociation constants for the WT and mutant\n",
      "EGFR kinases\n",
      "KinaseKd,n M Kd/Km[ATP] ,/H1100310/H110023\n",
      "Geﬁtinib AEE788 Geﬁtinib AEE788\n",
      "WT 35.3 /H110060.4 5.3 /H110060.3 6.8 1.0\n",
      "T790M 4.6 /H110060.1 27.6 /H110060.7 0.78 4.7\n",
      "L858R 2.4 /H110060.1 1.1 /H110060.1 0.016 0.0074\n",
      "L858R/T790M 10.9 /H110060.6 18.6 /H110060.5 1.3 2.2\n",
      "The ratio Kd/Km[ATP] provides a relative estimate of inhibitor potency.\n",
      "2072 /H20841www.pnas.org /H20862cgi /H20862doi /H2086210.1073 /H20862pnas.0709662105 Yun et al.\n",
      "crease inhibitor sensitivity at cellular concentrations of ATP.\n",
      "The expected potency of gefitinib (calculated Kiapp) is plotted as\n",
      "a function of ATP concentration for the L858R and L858R/T790M mutants in Fig. 3 A. Whereas the L858R mutant main-\n",
      "tains low-nanomolar sensitivity to gefitinib at cellular ATPconcentrations ( /H110151 mM), the L858R/T790M mutant does not.\n",
      "This predicted loss of inhibitor sensitivity in the L858R/T790Mmutant is confirmed by direct in vitro measurement of enzyme\n",
      "inhibition by gefitinib at 10\n",
      "/H9262M vs. 1 mM concentrations of ATP.\n",
      "The L858R/T790M mutant is sensitive to gefitinib at 10 /H9262M\n",
      "ATP, but resistant at 1 mM, whereas the L858R mutant iseffectively inhibited even at the higher concentration, whichapproximates the cellular level of ATP (Fig. 3 BandC). We also\n",
      "observe this effect in other assays using polyE\n",
      "4Y or the signaling\n",
      "adapter Shc as an EGFR substrate and in the presence of eitherMn\n",
      "2/H11001or Mg2/H11001(data not shown). We conclude that the clinically\n",
      "observed resistance of L858R/T790M mutant stems largely fromits enhanced affinity for ATP (as compared with the inhibitor-\n",
      "sensitive L858R mutant) and not from a steric block of inhibitorbinding as previously hypothesized.\n",
      "Discussion\n",
      "The present work highlights the extent to which the compro-\n",
      "mised ATP affinity of the EGFR mutants renders them suscep-tible to inhibition (at least for the L858R, G719S, and exon19deletions that have been studied to date) (8, 10). The T790Mmutation merely restores ATP affinity to the level of the WTkinase. In effect, the diminished ATP affinity of the oncogenicmutants open a ‘‘therapeutic window,’’ which renders them moreeasily inhibited relative to the WT EGFR and other kinases onwhich the inhibitors might have activity. The T790M secondarymutation effectively closes this window by restoring ATP affinityto WT levels.\n",
      "The activating nature of the gatekeeper mutation is not\n",
      "unique to EGFR, as indicated by its effect on Src and itspresence in chronic myelogenous leukemia before imatinibtreatment. The K\n",
      "meffect may be more idiosyncratic; note that\n",
      "the T790M mutation has little effect on Kmin the context of\n",
      "the WT EGFR kinase (Table 2). For the Abl T315I mutant, wemeasure K\n",
      "m[ATP] /H110051.8/H9262M as compared with Km[ATP] /H110056/H9262M\n",
      "for the WT Abl kinase (data not shown), this change in ATPaffinity is expected to have only a modest effect on inhibitorpotency. Also, it is clear that the T315I gatekeeper mutationin BCR-Abl directly blocks binding of imatinib and othercompounds with a similar binding mode.\n",
      "Fig. 2. Crystal structures of the EGFR T790M mutant show that inhibitors are readily accommodated in the active and inactive conformations of the kinase.\n",
      "(A) Superposition of EGFR T790M/AEE788 complex (yellow) and WT/AEE788 complex [light blue; drawn from PDB ID code 2J6M (8)]. Dashed lines indicate\n",
      "hydrogen bonds to the kinase hinge region that are preserved in both complexes. The location of the T790M mutation is indicated. ( B) Superposition of EGFR\n",
      "T790M/AEE788 complex (yellow) and apo-T790M structure (green). Note the alternate side-chain conformation of Met-790 in the presence of the inhibi tor. ( C)\n",
      "Crystal structure of HKI-272 in complex with the T790M mutant. The kinase adopts an inactive conformation, with the C-helix displaced. A covalent bon d is formed\n",
      "between Cys-797 and the crotonamide Michael acceptor of HKI-272. ( D) The structure of the T790M mutant in complex with HKI-272 (yellow) is superimposed\n",
      "on the structure of the WT EGFR kinase in complex with Lapatinib [light blue; drawn from PDB ID code 1XKK (32)]. In both structures, the kinase adopts the same\n",
      "inactive conformation and the inhibitors bind in a similar manner, with a single hydrogen bond to the hinge (dashed lines) and with their aniline subst ituents\n",
      "extending into the enlarged hydrophobic pocket that is characteristic of the inactive conformation.\n",
      "Table 2. Enzyme kinetic parameters of WT and mutant EGFR\n",
      "kinases\n",
      "KinaseKm[ATP],\n",
      "/H9262M kcat, s/H110021kcat/Km[ATP],\n",
      "/H9262M/H110021/H18528s/H110021\n",
      "WT 5.2 /H110060.2 0.026 5.00E-3\n",
      "T790M 5.9 /H110060.1 0.137 2.32E-2\n",
      "L858R 148 /H110064 1.484 1.00E-2\n",
      "L858R/T790M 8.4 /H110060.3 0.456 5.43E-2\n",
      "Yun et al. PNAS /H20841February 12, 2008 /H20841vol. 105 /H20841no. 6 /H208412073\n",
      "MEDICAL SCIENCES\n",
      "Our findings explain the puzzling observation that irreversible\n",
      "anilinoquinazoline inhibitors (and closely related irreversiblecompounds such as HKI-272) maintain efficacy against theT790M resistance mutant, and additionally they are importantfor understanding the nature of resistance and possible avenuesto the development of more effective drugs. The fact that theT790M substitution confers resistance by increasing the affinityfor ATP, rather than by simply sterically interfering with inhib-itor binding, means that T790M is a ‘‘generic’’ resistance mutant;it will tend to confer resistance to any ATP competitive inhibitor.To our knowledge, this mechanism of drug resistance, resistanceconferred by a mutation that increases affinity for a competingphysiologic substrate, has not been previously documented in aclinical context. Interestingly, a distinct, but related, effect hasrecently been described in a mutant of the mitotic kinesin KSP(also called Eg5). The KSP mutant was discovered in a labora-tory screen and confers drug resistance by an allosteric mech-anism involving enhanced affinity for ATP (33).\n",
      "As a class, irreversible inhibitors can overcome T790M resis-\n",
      "tance through covalent binding; once covalently bound, they areno longer in a competitive, reversible equilibrium with ATP. Anumber of such compounds are currently in clinical trials inoncology, including HKI-272, but none have yet received ap-proval. One concern with covalent inhibitors is the potential fortoxicity caused by off-target effects. At least 10 kinases inaddition to EGFR have a reactive cysteine residue in the positionequivalent to Cys-797 in EGFR, so it will be important tounderstand the activity of available irreversible agents againstthese kinases in particular, which include Tec family kinases,JAK3, and other kinases important for hematopoietic develop-ment and immune function. However, our results indicate thatirreversible binding is not required for effective inhibition of theT790M mutant. A reversible inhibitor that binds with sufficientaffinity to outcompete ATP should work as well. Calculationssimilar to those shown in Fig. 3 Aindicate that reversible\n",
      "inhibitors with affinity of /H11015200 pM or tighter against the T790M\n",
      "mutant should be effective.\n",
      "Methods\n",
      "Protein Preparation and Crystallization. Constructs spanning residues 696-1022\n",
      "of the human EGFR and bearing the WT sequence or the T790M and L858Rmutations were expressed and puriﬁed by using a baculovirus/insect cellsystem as described (8). Crystals of the T790M mutant were obtained in 0.1 MHepes (pH 7.5), 21% PEG6000, 0.3 M NaCl, and 5 mM tris(2-carboxyethyl)-phosphine (TCEP), whereas the T790M/HKI-272 complex crystals were made bycocrystallization in 0.1 M Hepes (pH 7.0), 0.2 M Li\n",
      "2SO4, 28% PEG3350, and 5\n",
      "mM TCEP. T790M/AEE788 complex crystals were made by soaking the apo-T790M crystals in 300\n",
      "/H9262M AEE788 inhibitor overnight.\n",
      "Structure Determination and Refinement. Diffraction data were collected at\n",
      "the Argonne National Laboratory (Argonne, IL) APS ID24 or ID19 beamlines at100 K. The data were processed with HKL2000 (34). The structures were solvedby molecular replacement method with PHASER (35), using the EGFR 696-1022G719S structure [Protein Data Bank (PDB) ID code 2itn] (8) for apo-T790M andT790M/AEE788 structures and EGFR 696-1022 V948R structure (PDB code 2gs7)(9) for the T790M/HKI-272 structure. CNS/simulated-annealing (36) was thenused to obtain less biased 2 F\n",
      "o/H11002FcandFo/H11002Fcmaps for manual inspection and\n",
      "adjustment of the model. Repeated rounds of manual reﬁtting and crystal-lographic reﬁnement were performed by using COOT (37) and refmac5 (38).Inhibitors were modeled into the closely ﬁtting positive F\n",
      "o/H11002Fcelectron\n",
      "density and then included in the following reﬁnement and ﬁtting cycles.Topology and parameter ﬁles for the inhibitors were generated by usingPRODRG (39).\n",
      "Enzyme Kinetic Assays, Inhibition Assays, and Data Analysis. EGFR kinetic\n",
      "parameters were determined in triplicate by using the ATP/NADH coupledassay system in a 96-well format as described (8). The reaction mixture con-tained 0.5 mg/ml BSA, 2 mM MnCl\n",
      "2, 1 mM phospho(enol) pyruvic acid (PEP;\n",
      "Sigma-Aldrich; catalogue no. P7002), 1 mM TCEP, 0.1 M MOPS 7.5, 5 mMpoly-[Glu\n",
      "4Tyr1] peptide (Sigma-Aldrich; catalogue no. P7244), 1/50 of the ﬁnal\n",
      "reaction mixture volume of pyruvate kinase/lactic dehydrogenase enzymes\n",
      "Fig. 3. The drug resistance of T790M secondary mutation is manifested only\n",
      "at cellular concentrations of ATP. ( A) The calculated Kiappfor the L858R single\n",
      "mutant and L858R/T790M double mutant are plotted versus ATP concentra-tion, using experimentally measured values for the K\n",
      "m[ATP] (Table 2) and\n",
      "setting Ki/H11005Kdas measured (Table 1). Note the expected loss of potency of the\n",
      "double mutant (solid line) as ATP concentrations approach cellular levels ( /H110151\n",
      "mM). ( B) Inhibition of L858R mutant EGFR kinase by geﬁtinib in the presence\n",
      "of 10/H9262M (black squares) or 1.0 mM ATP (red circles). ( C) Inhibition of the\n",
      "L858R/T790M double mutant EGFR by geﬁtinib in the presence of 10 /H9262M (black\n",
      "squares) or 1.0 mM ATP (red circles). In BandC, the in vitro kinase activity of\n",
      "the indicated EGFR mutant was measured in the presence of the indicatedconcentrations of geﬁtinib by using an EGFR Tyr-1173 autophosphorylationsite peptide (ENAEYLRVA) as substrate.\n",
      "2074 /H20841www.pnas.org /H20862cgi /H20862doi /H2086210.1073 /H20862pnas.0709662105 Yun et al.\n",
      "from rabbit muscle (Sigma-Aldrich; catalogue no. P-0294), 0.5 mM NADH, and\n",
      "0.5/H9262M EGFR kinase; ATP at varied concentration was added last to start the\n",
      "reaction. Steady-state initial velocity data were drawn from the slopes of theA\n",
      "340curves and ﬁt to the Michaelis-Menten equation to determine VmandKm\n",
      "values. To assure that our derived kcatparameters reﬂected concentrations of\n",
      "active enzyme, we estimated the active enzyme concentration of every kinasepreparation by titration of the samples with the tight binding inhibitorgeﬁtinib or AEE788 (see below).\n",
      "Inhibition assays were carried out by using the same kinetic assay method,\n",
      "with 10 mM MgCl\n",
      "2and 1.25 mM EGFR autophosphorylation site peptide\n",
      "(ENAEYLRVA) as the phospho-acceptor substrate. The ATP concentration wasﬁxed at 10\n",
      "/H9262M or 1 mM, and the indicated concentrations of the inhibitors\n",
      "were added before the addition of ATP.\n",
      "Binding Constant Assay and Calculation of the KiappValues. The equilibrium\n",
      "ﬂuorescence quenching method was used to obtain the binding constant andestimate the active enzyme concentration as described (8). Fluorescence mea-surements were carried out in a nitrogen-sparged buffer containing 20 mM\n",
      "Tris, 0.5% glycerol, 250 mM NaCl, and 1 mM TCEP. The obtained K\n",
      "dvalues and\n",
      "Km,ATP were used to calculate the Kiappvalues using the following equation\n",
      "(40):\n",
      "Kiapp/H11005Ki/H208491/H11001/H20851ATP/H20852/Km,ATP/H20850,\n",
      "assuming that the Kdvalues obtained in the binding assays are equal to Ki\n",
      "under the condition of the above kinetic assays.\n",
      "ACKNOWLEDGMENTS. We thank G. Caravotti (Novartis) and S. Rabindran\n",
      "(Wyeth Pharmaceuticals) for the compounds AEE788 and HKI-272, respec-tively; R. Copeland and C. Walsh for comments on the manuscript; and Y. Liand F. Poy for technical assistance. This work was supported by NationalInstitutes of Health Grants CA080942 (to M.J.E.) and CA116020 (to M.M.).M.J.E. is the recipient of a Scholar Award form the Leukemia and LymphomaSociety.\n",
      "1. Paez JG, et al. (2004) EGFR mutations in lung cancer: Correlation with clinical response\n",
      "to geﬁtinib therapy. Science 304:1497–1500.\n",
      "2. Lynch TJ, et al. (2004) Activating mutations in the epidermal growth factor receptor\n",
      "underlying responsiveness of non-small-cell lung cancer to geﬁtinib. N Engl J Med\n",
      "350:2129–2139.\n",
      "3. Pao W, et al. (2004) EGF receptor gene mutations are common in lung cancers from\n",
      "‘‘never smokers’’ and are associated with sensitivity of tumors to geﬁtinib and erlo-tinib. Proc Natl Acad Sci USA 101:13306–13311.\n",
      "4. Johnson BE, Janne PA (2005) Epidermal growth factor receptor mutations in patients\n",
      "with non-small cell lung cancer. Cancer Res 65:7525–7529.\n",
      "5. Gazdar AF, Shigematsu H, Herz J, Minna JD (2004) Mutations and addiction to EGFR:\n",
      "The Achilles ‘‘heal’’ of lung cancers? Trends Mol Med 10:481–486.\n",
      "6. Chan SK, Gullick WJ, Hill ME (2006) Mutations of the epidermal growth factor receptor\n",
      "in non-small cell lung cancer: Search and destroy. Eur J Cancer 42:17–23.\n",
      "7. Shigematsu H, Gazdar AF (2006) Somatic mutations of epidermal growth factor\n",
      "receptor signaling pathway in lung cancers. Int J Cancer 118:257–262.\n",
      "8. Yun CH, et al. (2007) Structures of lung cancer-derived EGFR mutants and inhibitor\n",
      "complexes: Mechanism of activation and insights into differential inhibitor sensitivity.Cancer Cell 11:217–227.\n",
      "9. Zhang X, Gureasko J, Shen K, Cole PA, Kuriyan J (2006) An allosteric mechanism for\n",
      "activation of the kinase domain of epidermal growth factor receptor. Cell125:1137–\n",
      "1149.\n",
      "10. Carey KD, et al. (2006) Kinetic analysis of epidermal growth factor receptor somatic\n",
      "mutant proteins shows increased sensitivity to the epidermal growth factor receptortyrosine kinase inhibitor, erlotinib. Cancer Res 66:8163–8171.\n",
      "11. Sordella R, Bell DW, Haber DA, Settleman J (2004) Geﬁtinib-sensitizing EGFR mutations\n",
      "in lung cancer activate antiapoptotic pathways. Science 305:1163–1167.\n",
      "12. Sharma SV, et al. (2006) A common signaling cascade may underlie ‘‘addiction’’ to the\n",
      "Src, BCR-ABL, and EGF receptor oncogenes. Cancer Cell 10:425–435.\n",
      "13. Pao W, et al. (2005) Acquired resistance of lung adenocarcinomas to geﬁtinib or\n",
      "erlotinib is associated with a second mutation in the EGFR kinase domain. PLoS Med\n",
      "2:e73.\n",
      "14. Kwak EL, et al. (2005) Irreversible inhibitors of the EGF receptor may circumvent\n",
      "acquired resistance to geﬁtinib. Proc Natl Acad Sci USA 102:7665–7670.\n",
      "15. Kobayashi S, et al. (2005) EGFR mutation and resistance of non-small-cell lung cancer\n",
      "to geﬁtinib. N Engl J Med 352:786–792.\n",
      "16. Kosaka T, et al. (2006) Analysis of epidermal growth factor receptor gene mutation in\n",
      "patients with non-small cell lung cancer and acquired resistance to geﬁtinib. Clin\n",
      "Cancer Res 12:5764–5769.\n",
      "17. Balak MN, et al. (2006) Novel D761Y and common secondary T790M mutations in\n",
      "epidermal growth factor receptor-mutant lung adenocarcinomas with acquired resis-tance to kinase inhibitors. Clin Cancer Res 12:6494–6501.\n",
      "18. Greulich H, et al. (2005) Oncogenic transformation by inhibitor-sensitive and -resistant\n",
      "EGFR mutants. PLoS Med 2:e313.\n",
      "19. Carter TA, et al. (2005) Inhibition of drug-resistant mutants of ABL, KIT, and EGF\n",
      "receptor kinases. Proc Natl Acad Sci USA 102:11011–11016.\n",
      "20. Sequist LV (2007) Second-generation epidermal growth factor receptor tyrosine kinase\n",
      "inhibitors in non-small cell lung cancer. Oncologist 12:325–330.21. Bell DW, et al. (2005) Inherited susceptibility to lung cancer may be associated with the\n",
      "T790M drug resistance mutation in EGFR. Nat Genet 37:1315–1316.\n",
      "22. Godin-Heymann N, et al. (2007) Oncogenic activity of epidermal growth factor recep-\n",
      "tor kinase mutant alleles is enhanced by the T790M drug resistance mutation. Cancer\n",
      "Res67:7319–7326.\n",
      "23. Regales L, et al. (2007) Development of new mouse lung tumor models expressing EGFR\n",
      "T790M mutants associated with clinical resistance to kinase inhibitors. PLoS ONE\n",
      "2:e810.\n",
      "24. Li D, et al. (2007) Bronchial and peripheral murine lung carcinomas induced by\n",
      "T790M-L858R mutant EGFR respond to HKI-272 and rapamycin combination therapy.Cancer Cell 12:81–93.\n",
      "25. Kwak EL, et al. (2006) Epidermal growth factor receptor kinase domain mutations in\n",
      "esophageal and pancreatic adenocarcinomas. Clin Cancer Res 12:4283–4287.\n",
      "26. Shah NP, et al. (2002) Multiple BCR-ABL kinase domain mutations confer polyclonal\n",
      "resistance to the tyrosine kinase inhibitor imatinib (STI571) in chronic phase and blastcrisis chronic myeloid leukemia. Cancer Cell 2:117–125.\n",
      "27. Roche-Lestienne C, et al. (2002) Several types of mutations of the Abl gene can be\n",
      "found in chronic myeloid leukemia patients resistant to STI571, and they can pre-existto the onset of treatment. Blood 100:1014–1018.\n",
      "28. Kato JY, et al. (1986) Amino acid substitutions sufﬁcient to convert the nontransform-\n",
      "ing p60c-src protein to a transforming protein. Mol Cell Biol 6:4155–4160.\n",
      "29. Yuza Y, et al. (2007) Allele-dependent variation in the relative cellular potency of\n",
      "distinct EGFR inhibitors. Cancer Biol Ther 6:661–667.\n",
      "30. Tsou HR, et al. (2005) Optimization of 6,7-disubstituted-4-(arylamino)quinoline-3-\n",
      "carbonitriles as orally active, irreversible inhibitors of human epidermal growth factorreceptor-2 kinase activity. J Med Chem 48:1107–1131.\n",
      "31. Stamos J, Sliwkowski MX, Eigenbrot C (2002) Structure of the epidermal growth factor\n",
      "receptor kinase domain alone and in complex with a 4-anilinoquinazoline inhibitor.J Biol Chem 277:46265–46272.\n",
      "32. Wood ER, et al. (2004) A unique structure for epidermal growth factor receptor bound\n",
      "to GW572016 (Lapatinib): Relationships among protein conformation, inhibitor off-rate, and receptor activity in tumor cells. Cancer Res 64:6652–6659.\n",
      "33. Luo L, et al. (2007) ATP-competitive inhibitors of the mitotic kinesin KSP that function\n",
      "via an allosteric mechanism. Nat Chem Biol 3:722–726.\n",
      "34. Otwinowski ZM, Minor W (1997) Processing of x-ray diffraction data collected in\n",
      "oscillation mode. Methods Enzymol 276:307–326.\n",
      "35. Read RJ (2001) Pushing the boundaries of molecular replacement with maximum\n",
      "likelihood. Acta Crystallogr D 57:1373–1382.\n",
      "36. Brunger AT, et al. (1998) Crystallography and NMR system: A new software suite for\n",
      "macromolecular structure determination. Acta Crystallogr D 54:905–921.\n",
      "37. Emsley P, Cowtan K (2004) Coot: Model-building tools for molecular graphics. Acta\n",
      "Crystallogr D 60:2126–2132.\n",
      "38. Murshudov GN, Vagin AA, Dodson EJ (1997) Reﬁnement of macromolecular structures\n",
      "by the maximum-likelihood method. Acta Crystallogr D 53:240–255.\n",
      "39. Schuttelkopf AW, van Aalten DM (2004) PRODRG: A tool for high-throughput crystal-\n",
      "lography of protein-ligand complexes. Acta Crystallogr D 60:1355–1363.\n",
      "40. Copeland RA (2000) Enzymes: A Practical Introduction to Structure, Mechanism, and\n",
      "Data Analysis (Wiley, New York), 2nd Ed, pp 305–317.\n",
      "Yun et al. PNAS /H20841February 12, 2008 /H20841vol. 105 /H20841no. 6 /H208412075\n",
      "MEDICAL SCIENCES\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create a variable that will select the selected number of pages\n",
    "pageobj=pdfreader.pages[x-1]\n",
    " \n",
    "#(x+1) because python indentation starts with 0.\n",
    "#create text variable which will store all text datafrom pdf file\n",
    "text=pageobj.extract_text()\n",
    " \n",
    "#save the extracted data from pdf to a txt file\n",
    "#we will use file handling here\n",
    "#dont forget to put r before you put the file path\n",
    "#go to the file location copy the path by right clicking on the file\n",
    "#click properties and copy the location path and paste it here.\n",
    "#put \"\\\\your_txtfilename\"\n",
    "file1=open(r\"C:\\Users\\ammar\\OneDrive\\Desktop\\phoenixbio\\Phoenix Bioinformatics\\test1.txt\",\"a\", encoding=\"utf-8\")\n",
    "file1.writelines(text)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
