{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import string\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from PmcidToText import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PMC8867171\n",
      "<Response [200]>\n",
      "https://www.ncbi.nlm.nih.gov//pmc/articles/PMC8867171/pdf/fpls-13-836519.pdf\n"
     ]
    }
   ],
   "source": [
    "IdScrape(\"PMC8867171\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plant Physiology, December 1999, Vol. 121, pp. 1207–1215, www.plantphysiol.org © 1999 American Society of Plant Physiologists\n",
      "\n",
      "Both phyA and phyB Mediate Light-Imposed Repression of\n",
      "\n",
      "PHYA Gene Expression in Arabidopsis1\n",
      "\n",
      "Francisco R. Canto´ n and Peter H. Quail*\n",
      "\n",
      "Department of Plant and Microbial Biology, University of California, Berkeley, California 94720 (F.R.C., P.H.Q.);\n",
      "\n",
      "and Plant Gene Expression Center/United States Department of Agriculture-Agricultural Research Service,\n",
      "\n",
      "800 Buchanan Street, Albany, California 94710 (F.R.C., P.H.Q.)\n",
      "\n",
      "The negatively photoregulated PHYA gene has a complex pro-\n",
      "moter structure in Arabidopsis, with three active transcription start\n",
      "sites. To identify the photoreceptors responsible for regulation of\n",
      "this gene, and to assess the relative roles of the three transcription\n",
      "start sites, we analyzed the changes in PHYA transcript levels in\n",
      "wild-type and photoreceptor mutant seedlings under various irra-\n",
      "diation conditions. Continuous far-red or red light exposures each\n",
      "induced a significant decline in transcript levels in wild-type etio-\n",
      "lated seedlings. Analysis of mutants specifically lacking either phyA\n",
      "or phyB protein demonstrated that these phytochromes are re-\n",
      "quired for the negative regulation induced by far-red and red light,\n",
      "respectively. Ribonuclease protection experiments showed further\n",
      "that this negative regulation is confined almost exclusively to the\n",
      "shortest, most abundant PHYA transcript, and occurs predomi-\n",
      "nantly in shoots. By contrast, both of the other minor transcripts in\n",
      "shoots, and all three transcripts in roots, exhibit near constitutive\n",
      "expression. This complex expression pattern indicates that the\n",
      "PHYA gene is subject to regulation by multiple signals, including\n",
      "environmental, developmental, and organ-specific signals.\n",
      "\n",
      "In higher plants, phytochromes are a family of photore-\n",
      "ceptor chromoproteins that monitor the red (R)/far-red\n",
      "(FR) region of the sunlight spectrum and regulate many\n",
      "photomorphogenic responses at all stages of the life cycle\n",
      "(Kendrick and Kronenberg, 1994). The phytochrome apo-\n",
      "protein is encoded by a small family of divergent genes,\n",
      "with five members (PHYA–PHYE) identified in Arabidop-\n",
      "sis (Sharrock and Quail, 1989; Clack et al., 1994). Each type\n",
      "of phytochrome holoprotein exists in two interconvertible\n",
      "conformations: the inactive, R-absorbing form (Pr) and the\n",
      "active FR-absorbing form (Pfr). High levels of phyA are\n",
      "present in etiolated seedlings and after prolonged periods\n",
      "of darkness, whereas the abundance of this molecular spe-\n",
      "cies decreases drastically and rapidly following illumina-\n",
      "tion with R or white (W) light. The photoregulation of\n",
      "phyA abundance occurs both at the protein level, by a\n",
      "rapid degradation of the unstable Pfr form, and at the\n",
      "\n",
      "1 This work was supported by grants from the National Science\n",
      "Foundation (no. MCB 9513590); the California Agricultural Exper-\n",
      "iment Station Project (no. CA–B*–PLB–5703 H); U.S. Department of\n",
      "Agriculture, Current Research\n",
      "(no.\n",
      "5335–21000 – 0010 – 00D); and the Spanish Ministry of Science and\n",
      "Education Fellowship to F.R.C.\n",
      "\n",
      "Information\n",
      "\n",
      "Service\n",
      "\n",
      "* Corresponding author; e-mail quail@nature.berkeley.edu; fax\n",
      "\n",
      "510 –559 –5678.\n",
      "\n",
      "1207\n",
      "\n",
      "At\n",
      "\n",
      "the functional\n",
      "\n",
      "mRNA level, by a light-induced reduction in PHYA tran-\n",
      "script abundance (Quail, 1994). In contrast, the abundance\n",
      "of PHYB, PHYC, PHYD, and PHYE transcripts are rela-\n",
      "tively unaltered by light in Arabidopsis (Clack et al., 1994).\n",
      "level, studies with phytochrome-\n",
      "deficient mutants have shown that phyA and phyB have\n",
      "distinct but overlapping roles in controlling seedling pho-\n",
      "tomorphogenesis (Reed et al., 1994; Quail et al., 1995;\n",
      "Whitelam and Devlin, 1997). phyA is predominantly, if not\n",
      "exclusively, responsible for de-etiolation in continuous far-\n",
      "red (FRc) light, whereas phyB is predominantly responsible\n",
      "for de-etiolation in response to continuous red (Rc) light.\n",
      "It has been shown in a variety of plant species that light\n",
      "induces an increase in the abundance of many different\n",
      "mRNAs, including those encoding the small subunit of\n",
      "Rubisco and chlorophyll a/b-binding protein (Silverthorne\n",
      "and Tobin, 1984), chloroplastic Gln synthetase (Edwards\n",
      "and Coruzzi, 1989) and Fd (Dobres et al., 1987), and others\n",
      "(Tobin and Kehoe, 1994; Terzaghi and Cashmore, 1995).\n",
      "Only a few identified genes have been demonstrated to be\n",
      "down-regulated in their expression by light,\n",
      "including\n",
      "genes encoding protochlorophyllide reductase (Forreiter et\n",
      "al., 1990), Asn synthetase (Tsai and Coruzzi, 1990), and\n",
      "PHYA itself (Colbert et al., 1983). The involvement of phy-\n",
      "tochromes in the light-induced changes in expression of\n",
      "some of these genes has been shown by reversibility of the\n",
      "effect of a single pulse of R light on the level of expression\n",
      "by a subsequent pulse of FR light, suggesting that this\n",
      "response has, at least, a low fluence response component\n",
      "(LFR). However, other genes show different fluence re-\n",
      "quirements (Terzaghi and Cashmore, 1995). For some of\n",
      "these light-regulated genes, including the oat PHYA gene,\n",
      "it has been demonstrated that phytochrome regulates their\n",
      "expression at the transcriptional level (Silverthorne and\n",
      "Tobin, 1987; Lissemore and Quail, 1988; Terzaghi and\n",
      "Cashmore, 1995).\n",
      "\n",
      "All PHYA genes that have been investigated are strongly\n",
      "expressed in the dark and negatively regulated by light\n",
      "(Quail, 1991). However, significant differences have been\n",
      "shown between monocots and dicots: (a) the extent of\n",
      "light-induced down-regulation varies substantially be-\n",
      "tween plant species, with the monocots in general appear-\n",
      "ing to respond more strongly than the dicots (Quail, 1994);\n",
      "(b) whereas monocot PHYA genes have only a single tran-\n",
      "scription start site (Hershey et al., 1987; Christensen and\n",
      "Quail, 1989; Kay et al., 1989), the dicot PHYA genes exam-\n",
      "\n",
      "\f1208\n",
      "\n",
      "Canto´ n and Quail\n",
      "\n",
      "Plant Physiol. Vol. 121, 1999\n",
      "\n",
      "ined, including that from Arabidopsis, have three tran-\n",
      "scription start sites (Sato, 1988; Dehesh et al., 1994; Adam et\n",
      "al., 1995), with evidence for differential control by phyto-\n",
      "chrome of the abundance of the transcripts initiated at each\n",
      "site in pea and tobacco (Tomizawa et al., 1989; Adam et al.,\n",
      "1995). The very low fluence (VLF) component shown by the\n",
      "light-imposed repression of the PHYA gene in oat suggests\n",
      "the action of at least phyA in this monocot (Quail, 1994). On\n",
      "the other hand, photobiological data have been interpreted\n",
      "as indicating that a stable phytochrome regulates the PHYA\n",
      "mRNA levels in pea seedlings through a LFR (Furuya et al.,\n",
      "1991), and a similar LFR pattern has been reported for\n",
      "PHYA mRNA levels in tobacco (Adam et al., 1994). These\n",
      "data raise the question of whether different phytochromes\n",
      "might be mediating this response in monocots and dicots.\n",
      "We wished to define the photoregulation of the PHYA\n",
      "gene in Arabidopsis, with the intent of using the genetic\n",
      "advantages of this model organism to dissect the molecular\n",
      "mechanisms involved. Although single pulses of R light\n",
      "did not decrease PHYA mRNA levels in etiolated Arabi-\n",
      "dopsis and tomato seedlings, a significant reduction in\n",
      "transcript levels was detected after continuous irradiation\n",
      "with white light (Wc) (Sharrock et al., 1988; Sharrock and\n",
      "Quail, 1989). Moreover, the negative photoregulation of a\n",
      "PHYA-GUS transgene by Rc and FRc irradiation in Arabi-\n",
      "dopsis seedlings suggests that at least one light-stable phy-\n",
      "tochrome and phyA may mediate light regulation of PHYA\n",
      "gene expression in this species (Somers and Quail, 1995).\n",
      "We determined the role of phyA and phyB in mediating\n",
      "negative photoregulation of the PHYA gene in Arabidopsis\n",
      "using phyA and phyB null mutants and have quantitatively\n",
      "defined the kinetics and fluence-rate dependence of FRc-\n",
      "light-imposed repression of this gene. In addition, we ex-\n",
      "amined the levels of the three nested PHYA transcripts in\n",
      "whole seedlings, shoots, and roots to determine their rela-\n",
      "tive contributions to the light-induced changes in total\n",
      "transcript levels and whether there is organ-specific mod-\n",
      "ulation of this response.\n",
      "\n",
      "MATERIALS AND METHODS\n",
      "\n",
      "Seedling Growth and Light Sources\n",
      "\n",
      "Arabidopsis ecotype RLD was used in all experiments,\n",
      "except that for the analysis of PHYA mRNA levels in a phyB\n",
      "null background,\n",
      "in which the mutant phyB-5 and the\n",
      "corresponding wild-type (WT) ecotype Landsberg erecta\n",
      "(Ler) were used.\n",
      "\n",
      "Seeds were surface-sterilized in 20% (v/v) commercial\n",
      "bleach and 0.03% (v/v) Triton X-100 for 10 min and then\n",
      "washed five times with sterile water. Seeds were plated on\n",
      "growth medium (Valvekens et al., 1988) without Suc. Ger-\n",
      "mination was induced by keeping the plates at 4°C in the\n",
      "dark for 2 d followed by 3-h exposure to white light at\n",
      "21°C. Plates were then returned to darkness at 21°C until\n",
      "the initiation of various continuous light treatments.\n",
      "\n",
      "FRc and Rc light were supplied with LED light sources\n",
      "(Quantum Devices, Barnveld, WI) and Wc light with cool-\n",
      "white fluorescent lamps. The fluence rates used routinely\n",
      "\n",
      "were 390 mmol m22 s21 for FRc and Rc irradiations and 39\n",
      "mmol m22 s21 for Wc light, unless otherwise stated.\n",
      "\n",
      "RNA Isolation, Analysis, and Quantification\n",
      "\n",
      "Total RNA was extracted as described previously (Can-\n",
      "to´n et al., 1993) or by using the RNasy purification kit from\n",
      "Qiagen (Valencia, CA).\n",
      "\n",
      "For northern-blot analysis, equal amounts of total RNA\n",
      "samples were denatured at 65°C for 15 min in 13 MOPS\n",
      "buffer (20 mm 3-[N-Morpholino] propanesulfonic acid, pH\n",
      "7.0, 1 mm EDTA, and 5 mm sodium acetate), 6.6% (v/v)\n",
      "formaldehyde, and 50% (v/v) formamide and separated by\n",
      "electrophoresis in 1.0% (w/v) agarose/6% (v/v) formalde-\n",
      "hyde/13 MOPS buffer gels. After electrophoresis the RNA\n",
      "was transferred to nylon membranes (MAGNA, Micron\n",
      "Separations, Westborough, MA). The membranes were pre-\n",
      "hybridized and hybridized at 65°C in the hybridization\n",
      "buffer described by Church and Gilbert (1984). Specific\n",
      "32P-labeled probes were synthesized by the Multiprime\n",
      "DNA labeling system (Amersham-Pharmacia Biotech,\n",
      "Uppsala) using as a template a 642-bp DNA fragment from\n",
      "the 39 end of an Arabidopsis PHYA cDNA clone (nucleo-\n",
      "tides [nt] 3,128–3,770) and a 262-bp DNA fragment from a\n",
      "PHYB cDNA clone (nt 3,607–3,868) (Sharrock and Quail,\n",
      "1989). As a loading control membranes were rehybridized\n",
      "with a 400-bp fragment of a 17S rRNA from rice (nt 158–\n",
      "557) (Takaiwa et al., 1984), which hybridizes specifically\n",
      "with the 18S RNA from Arabidopsis.\n",
      "\n",
      "For RNase protection experiments, a ScaI-KpnI (position\n",
      "2259 to 1125) fragment from the 59-UTR and promoter\n",
      "region of the PHYA gene was cloned into the pBluescript\n",
      "plasmid (Stratagene, La Jolla, CA). Labeled antisense RNA\n",
      "was produced by in vitro transcription using the linearized\n",
      "plasmid as a template and [a-32P]CTP to generate a probe\n",
      "430 nt in length, in which the first 12 nt at the 59 and the last\n",
      "38 nt at the 39 ends are derived from plasmid sequence. The\n",
      "RNase assay was performed using the RPA II Kit (Ambion,\n",
      "Austin, TX) following the instructions of the manufacturer.\n",
      "RNA size markers were obtained by in vitro transcription\n",
      "of the Century Marker template (Ambion). The expected\n",
      "sizes of the protected fragments from the PHYA transcripts\n",
      "were 121 nt for mRNA1, 204 nt for mRNA2, and 380 nt for\n",
      "mRNA3.\n",
      "\n",
      "For quantitative analysis, hybridized blots and RNase\n",
      "protection gels were exposed to phosphor imager plates\n",
      "and images were subsequently obtained and analyzed with\n",
      "a Storm 860 system and ImageQuant version 1.1 software\n",
      "(Molecular Dynamics, Sunnyvale, CA). The radioactivity\n",
      "values for each different-length RNase protection fragment\n",
      "were corrected for the number of cytidines present in that\n",
      "probe fragment to permit direct quantitative comparison of\n",
      "the abundance of mRNA1, mRNA2, and mRNA3.\n",
      "\n",
      "RESULTS\n",
      "\n",
      "To more definitively determine whether phytochromes\n",
      "do indeed regulate PHYA mRNA abundance in Arabidop-\n",
      "sis seedlings, and to identify the family members poten-\n",
      "tially involved, we analyzed the transcript steady-state\n",
      "\n",
      "\fRegulation of PHYA Expression in Arabidopsis\n",
      "\n",
      "1209\n",
      "\n",
      "levels from wild-type and mutant 5-d-old etiolated seed-\n",
      "lings and seedlings irradiated for 24 h before harvest with\n",
      "FRc, Rc, or Wc light (Fig. 1). For this purpose we used the\n",
      "mutants phyA-101 (Dehesh et al., 1993) and phyB-5 (Koorn-\n",
      "neef et al., 1980; Reed et al., 1993), and the relevant wild-\n",
      "type backgrounds RLD and Ler. These mutants are null for\n",
      "phyA or phyB, respectively, as a result of point mutations\n",
      "introducing early stop codons into the protein coding re-\n",
      "gions of the genes. As a result of the mutation in phyA-101,\n",
      "the overall levels of PHYA mRNA were lower relative to\n",
      "the wild type under all conditions (Fig. 1A, left), but were\n",
      "still detectable by northern-blot hybridization. The intensi-\n",
      "ties of radioactive signals were quantified and the results\n",
      "are summarized in Figure 1B.\n",
      "\n",
      "All light qualities elicited a reduction in PHYA mRNA\n",
      "levels in wild-type etiolated seedlings compared with non-\n",
      "irradiated seedlings (Fig. 1B, RLD and Ler). However, the\n",
      "phyA-101 mutant failed specifically to respond to FRc light,\n",
      "while still showing a response to Rc and Wc light. In\n",
      "contrast, the phyB null mutant had full responsiveness to\n",
      "FRc light (compare levels of mRNA in FRc in Ler and\n",
      "\n",
      "Figure 2. Fluence rate response curve of PHYA and PHYB mRNA\n",
      "levels in RLD seedlings irradiated with FRc light. A, Four-day-old\n",
      "dark-grown seedlings were exposed to increasing FRc fluence rates\n",
      "(0, 0.3, 1.2, 2.5, 4.7, 7.2, 30, 71, 200, and 385 mmol m22 s21) for a\n",
      "24-h period, and 4.5 mg of total RNA was analyzed by northern blot\n",
      "with specific PHYA, PHYB, and 18s rRNA probes and visualized by\n",
      "autoradiography. B, Relative PHYA and PHYB mRNA levels as a\n",
      "function of the fluence rate of FRc light. The relative transcript levels\n",
      "were expressed as a percentage of the corresponding value in dark\n",
      "after normalization to the 18S rRNA hybridization signal. The mean\n",
      "of two independent experiments was determined and the bars indi-\n",
      "cate the range of the two values.\n",
      "\n",
      "phyB-5), but lost most of its response to Rc and partially to\n",
      "Wc light. Therefore, the changes in the amount of PHYA\n",
      "mRNA levels induced by FRc and Rc light are indepen-\n",
      "dently mediated by different phytochromes. These data\n",
      "suggest that, as in the overall process of de-etiolation,\n",
      "phyA is necessary for changes in PHYA gene expression in\n",
      "FRc light, and no other phytochrome (phyB–E) seems to\n",
      "contribute to this response. In contrast, phyB is necessary\n",
      "for full control of PHYA gene expression in Rc light and no\n",
      "other phytochrome is able to account for this response to\n",
      "the same magnitude.\n",
      "\n",
      "We focused on the effect of FRc light on PHYA expres-\n",
      "sion, because this quality of light is non-photosynthetically\n",
      "\n",
      "Figure 1. Phytochromes A and B control PHYA mRNA levels. A,\n",
      "Northern-blot analysis of 8 mg of total RNA isolated from seedlings of\n",
      "a phyAnull mutant (phyA-101), a phyBnull mutant (phyB-5), and the\n",
      "corresponding wild-types RLD and Ler, respectively. Seedlings were\n",
      "grown for 5 d in continuous darkness (black bars) or transferred to\n",
      "FRc light (dark gray bars), Rc light (light gray bars), or Wc light (white\n",
      "bars) 24 h prior to harvest. The fluence rates used were 390 mmol\n",
      "m22 s21 for FRc and Rc light and 39 mmol m22 s21 for Wc light. B,\n",
      "PHYA transcript levels normalized to the 18S rRNA signal were\n",
      "expressed as a percentage of the corresponding value in the dark for\n",
      "each line. The mean of two completely independent experiments\n",
      "was determined, and the bars indicate the range of the two values.\n",
      "\n",
      "\f1210\n",
      "\n",
      "Canto´ n and Quail\n",
      "\n",
      "Plant Physiol. Vol. 121, 1999\n",
      "\n",
      "70% of the dark level). Therefore, the decrease in PHYA\n",
      "gene expression appears to be fluence rate dependent over\n",
      "a wide range of FRc fluence rates.\n",
      "\n",
      "To determine whether the photoresponsiveness of PHYA\n",
      "expression to FRc light varied with the developmental age\n",
      "of the seedlings, we examined the time-course of PHYA\n",
      "mRNA accumulation in etiolated seedlings and its decline\n",
      "over a 24-h illumination period in seedlings of increasing\n",
      "age (Fig. 3). The quantification data summarized in Figure\n",
      "3B show that the PHYA transcript levels increased during\n",
      "development of the seedlings in darkness at least up to 9 d\n",
      "old. However, irradiation of 4-d or older etiolated seed-\n",
      "lings with FRc light for 24 h before harvest suppressed the\n",
      "accumulation of PHYA transcript, reducing the steady-\n",
      "state levels to a similar value at all these stages. By contrast,\n",
      "3-d-old seedlings showed significantly lower relative re-\n",
      "sponsiveness to the FRc light treatment. We also observed\n",
      "differences between seed batches in the relative amount of\n",
      "PHYA mRNA accumulated in darkness, possibly because\n",
      "of differences in the rates of seedling development. The\n",
      "resultant variability in levels at the start of the irradia-\n",
      "\n",
      "Figure 3. Responsiveness of PHYAmRNA levels to FRc illumination\n",
      "in dark-grown seedlings of increasing age. Total RNA was isolated\n",
      "from seedlings grown for increasing numbers of days in continuous\n",
      "dark (D) or irradiated with FRc light for 24 h before harvesting (D 3\n",
      "Fc). A, Blot hybridization of 5 mg of total RNA with the PHYA and\n",
      "18S rRNA specific probes. B, Relative PHYAmRNA levels at increas-\n",
      "ing days after germination expressed as a percentage of the transcript\n",
      "amount in 3-d-old dark-grown seedlings after normalization to the\n",
      "18S rRNA hybridization signal. F, Non-irradiated seedlings; E, seed-\n",
      "lings irradiated for 24 h before harvesting.\n",
      "\n",
      "active and is a good indicator of phytochrome action. In\n",
      "addition, the results shown above for the regulation of\n",
      "PHYA gene expression are in agreement with the conclu-\n",
      "sion that only one molecular species of phytochrome\n",
      "(phyA) mediates the response to FRc light. Thus, this light\n",
      "quality provides us with the simplest situation involving a\n",
      "single photoreceptor and one responsive gene.\n",
      "\n",
      "First, we measured the steady-state levels of PHYA\n",
      "mRNA as a function of FRc fluence rate (Fig. 2). PHYB\n",
      "mRNA levels were also determined as a control. The accu-\n",
      "mulation of PHYA mRNA was inversely related to fluence\n",
      "rate, with the highest levels in seedlings maintained in\n",
      "complete darkness. Fluence rates as low as 0.4 mmol m22\n",
      "s21 induced a significant reduction in mRNA accumula-\n",
      "tion; however, maximal suppression of PHYA expression\n",
      "required FRc illumination of moderate intensity (70 mmol\n",
      "m22 s21). On the contrary, fluence rates below 10 mmol\n",
      "m22 s21 did not have a significant effect on PHYB mRNA\n",
      "steady-state levels and only illumination of moderate and\n",
      "high intensities decreased mRNA levels to some extent (to\n",
      "\n",
      "Figure 4. Time course of repression of PHYA mRNA levels in re-\n",
      "sponse to FRc irradiation. A, Northern blot of total RNA (2.5 mg)\n",
      "isolated from 7-d-old seedlings grown in continuous darkness and\n",
      "transferred to FRc light 0, 1, 3, 6, 12, and 24 h prior to harvest. As a\n",
      "control, samples were harvested at the same time points from seed-\n",
      "lings maintained in darkness (D). B, Relative PHYAmRNA levels are\n",
      "shown as a percentage of the value at time zero after normalization\n",
      "to the 18S rRNA hybridization signal. The mean of two independent\n",
      "experiments was determined and the bars indicate the range of the\n",
      "two values.\n",
      "\n",
      "\fRegulation of PHYA Expression in Arabidopsis\n",
      "\n",
      "1211\n",
      "\n",
      "Figure 5. Differential response of the three nested PHYA transcript levels to continuous light. Total RNA (8 mg) extracted\n",
      "from 5-d-old seedlings grown in continuous dark (D) or transferred to FRc, Rc, or Wc light 24 h prior to harvest were\n",
      "analyzed by RNase protection assay. A, Schematic of PHYApromoter and surrounding DNA and fragment used as a template\n",
      "for riboprobe synthesis. B, Autoradiograph: Ct, 8 mg of yeast RNA; P, full-length undigested probe; M, labeled molecular\n",
      "mass markers; asterisk, full-length undigested probe; nt., nucleotide. C, The hybridization signal for every protected fragment\n",
      "was quantified and corrected for the differences in length. The mean of two independent experiments was determined and\n",
      "the bars indicate the range of the two values.\n",
      "\n",
      "tion treatments could explain the differences in final rela-\n",
      "tive levels after irradiation observed between Figures 1,\n",
      "2, and 4.\n",
      "\n",
      "To determine how rapidly PHYA expression is sup-\n",
      "pressed in FRc light, we determined the short-term kinetics\n",
      "of PHYA mRNA levels in 7-d-old etiolated seedlings trans-\n",
      "ferred to FRc light for 24 h. As a control, PHYA mRNA\n",
      "levels were also monitored in seedlings maintained in\n",
      "darkness. Figure 4A shows the results of blot hybridization\n",
      "with specific PHYA and 18S rRNA probes for seedlings\n",
      "illuminated with FRc light or kept in darkness for the\n",
      "period indicated. The quantification of the hybridization\n",
      "signals showed a significant reduction in PHYA mRNA\n",
      "levels within 1 to 3 h following the transfer from dark to\n",
      "light, with only 30% of the zero-time dark levels being\n",
      "detected after 6 h of exposure to FRc light (Fig. 4B). Longer\n",
      "periods of illumination up to 24 h induced a relatively\n",
      "small further decrease in PHYA transcript levels, suggest-\n",
      "ing that the response was almost saturated after 6 h of\n",
      "continuous irradiation.\n",
      "\n",
      "We wished to determine the relative contributions of the\n",
      "three previously defined Arabidopsis PHYA transcripts\n",
      "(Dehesh et al., 1994) to the overall differences in PHYA\n",
      "mRNA levels between etiolated and irradiated seedlings.\n",
      "For this purpose, 4-d-old etiolated seedlings were exposed\n",
      "to different light qualities for 24 h and total RNA was\n",
      "\n",
      "analyzed by RNase protection assay using a [a-32P]CTP\n",
      "59-\n",
      "uniformly labeled riboprobe derived from the\n",
      "untranslated region of the Arabidopsis PHYA gene (Fig.\n",
      "5A). The autoradiograph (Fig. 5B) shows the three pro-\n",
      "tected fragments derived from the PHYA transcripts with\n",
      "the expected sizes (121, 204, and 380 nt) only in the samples\n",
      "from Arabidopsis seedlings. No similar bands were de-\n",
      "tected in the yeast RNA control. The extra fourth band at\n",
      "the top of all the samples (labeled with an asterisk), includ-\n",
      "ing the yeast RNA control, represents the full-length probe\n",
      "remaining after digestion with RNase.\n",
      "\n",
      "As shown in Figure 5C, the three transcripts contributed\n",
      "quantitatively to different extents to the total pool of PHYA\n",
      "mRNA in the seedlings. The shortest transcript (mRNA1)\n",
      "was the most abundant, accounting for most of the PHYA\n",
      "transcript levels in the dark, and the abundance of the\n",
      "other two transcripts decreased with length. In addition,\n",
      "most of the quantitative differences observed in PHYA\n",
      "mRNA abundance between irradiated and non-irradiated\n",
      "seedlings were the result of the decrease in mRNA1 levels.\n",
      "Although some minor changes were observed in the levels\n",
      "of mRNA2 and mRNA3, their relative contributions to the\n",
      "changes in the total pool of PHYA transcript levels were\n",
      "relatively insignificant, because their absolute levels were\n",
      "low in both conditions relative to mRNA1, and the effect of\n",
      "light on their relative levels was small.\n",
      "\n",
      "\f1212\n",
      "\n",
      "Canto´ n and Quail\n",
      "\n",
      "Plant Physiol. Vol. 121, 1999\n",
      "\n",
      "Figure 6. Differential photoresponsiveness of PHYA mRNA1 levels in shoots and roots of Arabidopsis seedlings. A, Total\n",
      "RNA (5 mg) from shoots and roots of 7-d-old seedlings grown either in darkness (D) or under 39 mmol m22 s21 Wc light were\n",
      "analyzed by RNase protection assay. Ct, Yeast RNA (5 mg), P, full-length undigested probe; M, labeled molecular mass\n",
      "marker. The size of the RNA markers are indicated in nucleotides on the right. The asterisk indicates full-length undigested\n",
      "probe. B, The hybridization signal for every protected fragment was quantified and corrected for the differences in length.\n",
      "The abundance of each transcript in shoot (top) and roots (bottom) is shown in the left panels. The right panels show the sum\n",
      "of the levels of the three transcripts in each treatment for shoots (top) and roots (bottom). Note that PHYA mRNA levels\n",
      "represent relative abundance per mass of total RNA (5 mg) extracted from each organ, so differences between root and shoot\n",
      "do not reflect comparative absolute abundance per organ.\n",
      "\n",
      "Figure 6 shows the analysis of PHYA transcript levels in\n",
      "total RNA extracted separately from shoots and roots of\n",
      "seedlings grown for 7 d completely in darkness or Wc light.\n",
      "Three protected fragments with the same sizes as in Figure\n",
      "5A were present in all the samples (Fig. 6A), indicating that\n",
      "the three transcripts are accumulated to some extent in\n",
      "both organs. Figure 6B shows quantitatively the relative\n",
      "abundance of each transcript in each sample (left) and the\n",
      "result of totaling the abundance of the three mRNAs\n",
      "(right). mRNA1 is clearly the most abundant transcript in\n",
      "the shoots of dark-grown seedlings. Although for the root\n",
      "samples mRNA1 appears visually to be less abundant than\n",
      "mRNA2 or mRNA3 (Fig. 6A), after integration of the ra-\n",
      "dioactive counts in the more diffuse mRNA1 band (actually\n",
      "three closely spaced bands), and correction for the number\n",
      "of labeled cytidines in the different-length probe frag-\n",
      "ments, mRNA1 was in fact the most abundant transcript in\n",
      "roots as well (see Fig. 6B, root).\n",
      "\n",
      "Like in the RNA samples isolated from whole seedlings,\n",
      "the main difference in the overall PHYA mRNA levels\n",
      "between shoots of dark-grown and light-grown seedlings\n",
      "was as a result of changes in mRNA1 levels (with Wc-light-\n",
      "grown shoots having 21% of the dark value; Fig. 6B, top).\n",
      "\n",
      "The differences in levels of mRNA2 and mRNA3 contrib-\n",
      "uted little to the overall changes in PHYA transcript abun-\n",
      "dance. Similarly, because mRNA1 was the most abundant\n",
      "transcript in roots, this was also the main transcript ac-\n",
      "counting for the overall differences in PHYA mRNA levels\n",
      "between darkness and light in this organ (Fig. 6B, bottom).\n",
      "However, the extent to which the levels of mRNA1 were\n",
      "reduced by light in roots was comparatively low (78% of\n",
      "the dark value) relative to shoots. As a result, the overall\n",
      "levels of PHYA transcripts were nearly the same in light-\n",
      "and dark-grown roots. It is important to note that the levels\n",
      "of mRNAs shown in shoots and roots represent relative\n",
      "abundance per mass of total RNA (5 mg) extracted from\n",
      "each organ. Thus, the levels shown do not correspond to\n",
      "absolute differences in PHYA transcript abundance be-\n",
      "tween the two organs in a single seedling because they are\n",
      "not directly comparable.\n",
      "\n",
      "DISCUSSION\n",
      "\n",
      "The decrease in PHYA mRNA levels in wild-type Arabi-\n",
      "dopsis seedlings in response to irradiation, not only with\n",
      "Wc light, but also with Rc light and FRc light (Fig. 1),\n",
      "\n",
      "\fRegulation of PHYA Expression in Arabidopsis\n",
      "\n",
      "1213\n",
      "\n",
      "suggests the involvement of one or more phytochromes.\n",
      "This observation is in close agreement with the previously\n",
      "reported photoregulation of a PHYA-GUS transgene in\n",
      "Arabidopsis (Somers and Quail, 1995), suggesting that the\n",
      "changes in the level of PHYA transcript shown here may\n",
      "be determined primarily by photocontrol of the transcrip-\n",
      "tion rate. Moreover, the data in Figure 1 provide genetic\n",
      "evidence that at least two molecular species of phyto-\n",
      "chromes (phyA and phyB) are involved in the light-\n",
      "induced decrease of PHYA mRNA levels and that, like in\n",
      "the regulation of de-etiolation, phyA and phyB mediate the\n",
      "response of PHYA gene expression to FRc and Rc light,\n",
      "respectively. The significant decrease of PHYA transcript\n",
      "levels under Wc light compared with that under Rc light in\n",
      "the phyB mutant suggests additional regulation by other\n",
      "photoreceptors, such as the cryptochromes.\n",
      "\n",
      "Previous photobiological studies using pulse R and FR\n",
      "irradiation were interpreted as indicating that PHYA ex-\n",
      "pression is regulated by phyA in monocots such as oats\n",
      "(Quail, 1994) and by a stable phytochrome in dicots such as\n",
      "pea and tobacco (Furuya et al., 1991; Adam et al., 1994). The\n",
      "present data were obtained under continuous irradiation\n",
      "conditions, indicating that both phyA and phyB can regu-\n",
      "late Arabidopsis PHYA expression via FR and R high-\n",
      "irradiance response modes, respectively (Fig. 1 and 2).\n",
      "Whether other dicots exhibit similar control of PHYA re-\n",
      "mains to be determined. Likewise, it remains to be seen\n",
      "whether monocots null for phyA would exhibit a cryptic R-\n",
      "high-irradiance response under phyB control, once the\n",
      "dominant influence of phyA has been removed.\n",
      "\n",
      "In terms of kinetics, the response of the Arabidopsis\n",
      "PHYA gene to FRc light (Fig. 4) is broadly similar to the\n",
      "response to Wc light in Cucurbita (Lissemore et al., 1987),\n",
      "pea (Sato, 1988), and the monocot oat (Colbert et al., 1985).\n",
      "In these four species the level of transcript reached a min-\n",
      "imum a few hours after starting the irradiation, suggesting\n",
      "a similar rate of degradation of the transcripts in the dif-\n",
      "ferent species. On the other hand, significant differences\n",
      "are observed in the extent of the decrease, with Arabidop-\n",
      "sis and other dicots showing a quantitatively lower extent\n",
      "of repression than the monocot.\n",
      "\n",
      "One caveat to this observation is illustrated in Figure 3B,\n",
      "which shows that the responsiveness of PHYA mRNA lev-\n",
      "els to FRc illumination in etiolated seedlings of increasing\n",
      "age in the range of 4 to 8 d was very similar in terms of\n",
      "absolute basal levels of transcript reached after the light\n",
      "treatment, but different in relative levels when expressed\n",
      "as a percentage of the corresponding dark level in seed-\n",
      "lings of the same age. This effect was even more pro-\n",
      "nounced in younger (3-d-old) seedlings (Fig. 3B). This pat-\n",
      "tern is reminiscent of\n",
      "the decline in spectroscopically\n",
      "detectable phytochrome in mustard seedlings cotyledons\n",
      "in response to FRc light, where the same stable plateau\n",
      "level was reached independent of the starting level (Scha¨fer\n",
      "et al., 1972). These observations indicate that the extent of\n",
      "repression observed relative to the dark levels depends on\n",
      "the stage of development examined, and raise the possibil-\n",
      "ity that absolute basal levels of PHYA transcript in light-\n",
      "grown plants could be more similar in monocots and dicots\n",
      "than was previously thought. The relatively small response\n",
      "\n",
      "of 3-d-old Arabidopsis seedlings to FRc light (Fig. 3B)\n",
      "emphasizes that the response is subject to developmental\n",
      "control.\n",
      "\n",
      "the contribution of\n",
      "\n",
      "The levels of the three mRNAs transcribed from the\n",
      "PHYA gene in Arabidopsis are differentially regulated by\n",
      "Rc, FRc, and Wc irradiation (Fig. 5). mRNA1 is the major\n",
      "in the dark and the most strongly light-\n",
      "transcript\n",
      "regulated.\n",
      "In contrast,\n",
      "the minor\n",
      "mRNA2 and mRNA3 to the overall changes in PHYA tran-\n",
      "script level is limited. The differential effect of the light\n",
      "treatments on the levels of the three PHYA transcripts\n",
      "could be the result of differences in the responsiveness of\n",
      "the transcriptional activities driven from the three different\n",
      "TATA boxes. Alternatively, light could differentially affect\n",
      "the stability of the three mRNAs, mainly altering the deg-\n",
      "radation rate of the shorter transcript. Although the deter-\n",
      "minants for RNA stability are considered to reside usually\n",
      "in the 39 end of transcripts (Gallie, 1993), differences in\n",
      "stability among the three PHYA mRNAs determined by\n",
      "elements present in the non-shared region of the 59-UTRs\n",
      "cannot be ruled out.\n",
      "\n",
      "The relative position of TATA box 1 in the Arabidopsis\n",
      "promoter, as well as a sequence with homology to the\n",
      "functionally defined cis repressor element (Dehesh et al.,\n",
      "1994), are very similar to those in the oat PHYA gene (Bruce\n",
      "et al., 1991), suggesting a similar basic promoter structure\n",
      "and repression mechanism in different\n",
      "species. The\n",
      "mRNA1s from the pea PHYA and the tobacco PHYA1\n",
      "genes are also the major transcripts in these species and the\n",
      "most strongly light regulated (Tomizawa et al., 1989; Adam\n",
      "et al., 1995). However, in the tobacco PHYA2 gene, mRNA2\n",
      "instead of mRNA1 is the major and most light-regulated\n",
      "transcript (Adam et al., 1995). A sequence similar to the\n",
      "repressor element identified in oat is also present in the\n",
      "PHYA promoter of pea (Sato, 1988; Dehesh et al., 1994), and\n",
      "the discovery that a sequence with homology to this\n",
      "element is also required for light-imposed repression of\n",
      "the Asn synthetase gene promoter from pea (Neuhaus\n",
      "et al., 1997; Ngai et al., 1997) suggests the possibility that\n",
      "this sequence is a common element mediating negative\n",
      "light regulation of plant promoters. However, the function-\n",
      "ality of this sequence in dicot PHYA genes remains to be\n",
      "determined.\n",
      "\n",
      "In particular,\n",
      "\n",
      "Previous work using transgenic Arabidopsis seedlings\n",
      "harboring a PHYA-GUS gene fusion showed differential\n",
      "light regulation of the PHYA promoter activity in shoots\n",
      "and roots (Somers and Quail, 1995). A differential organ-\n",
      "specific distribution of the three mRNAs could have ex-\n",
      "plained the differences in photoregulation of PHYA be-\n",
      "tween the two organs.\n",
      "the absence of\n",
      "expression of the light-regulated mRNA1 in roots could\n",
      "have potentially accounted for these observed differences.\n",
      "However, the three transcripts are accumulated to some\n",
      "extent in both organs, and mRNA1 is the major transcript\n",
      "in roots as well as shoots (Fig. 6B). Nevertheless, a signif-\n",
      "icant difference in the extent of the repression imposed by\n",
      "light on the levels of mRNA1 was observed between or-\n",
      "gans. Whereas only 21% of the dark value of mRNA1\n",
      "remained in Wc-light-grown shoots, the levels of this tran-\n",
      "script in roots were as high as 78% relative to dark-grown\n",
      "\n",
      "\f1214\n",
      "\n",
      "Canto´ n and Quail\n",
      "\n",
      "Plant Physiol. Vol. 121, 1999\n",
      "\n",
      "roots. These data suggest that the specific light control of\n",
      "expression driven from TATA box 1 of PHYA is strongly\n",
      "modulated by organ-specific factors.\n",
      "\n",
      "The presence of alternative promoters is postulated to\n",
      "allow more flexibility in the regulation of the gene. In\n",
      "higher eukaryotes, such multiple promoters are frequently\n",
      "associated with genes that are expressed in a tissue-specific\n",
      "and/or developmentally specific manner (Schibler and Si-\n",
      "erra, 1987; Ayoubi and Van De Ven, 1996; Myers et al.,\n",
      "1998; Holzfeind et al., 1999). In plants, differential devel-\n",
      "opmental regulation of alternative promoters in the nuclear\n",
      "gene encoding the chloroplast ribosomal protein L21 of\n",
      "spinach (Lagrange et al., 1993) and the use of two alterna-\n",
      "tive promoters from the chalcone flavanone isomerase A\n",
      "gene in various flower tissues of Petunia hybrida (Tunen et\n",
      "al., 1989) have been reported.\n",
      "\n",
      "the mRNA1 promoter unit\n",
      "\n",
      "To our knowledge, dicot PHYA is the first plant gene\n",
      "reported with multiple promoters that are differentially\n",
      "regulated by an environmental factor. In this gene the\n",
      "occurrence of two TATA boxes weakly regulated or not\n",
      "regulated by light allows a certain persistent level of PHYA\n",
      "expression in light. This could be required to maintain\n",
      "basal levels of phyA protein in the light, where it has an\n",
      "active role in fully de-etiolated plants, as indicated from\n",
      "physiological studies with phyA and phyB mutants (John-\n",
      "son et al., 1994; Reed et al., 1994; Whitelam and Devlin,\n",
      "1997). By contrast,\n",
      "is\n",
      "strongly expressed in the dark and negatively regulated in\n",
      "the light allows the modulation of PHYA expression in\n",
      "response to variable light conditions, and has the capacity\n",
      "to generate high levels of phyA during development in\n",
      "darkness or dark transitions. This pool of phyA generated\n",
      "in the dark and added to the basal pool may be critical for\n",
      "phyA function in dark-to-light transitions and FR-enriched\n",
      "environments in fully green plants. If this is so, it is possi-\n",
      "ble that the monocots evolved a different strategy to com-\n",
      "pensate for the absence of the two extra TATA boxes. On\n",
      "the other hand, we do not know how mRNA2 and mRNA3\n",
      "contribute to the cellular phyA protein pool in Arabidop-\n",
      "sis, and the presence of three additional short open reading\n",
      "frames in the 59-UTR of mRNA3 could diminish its rate of\n",
      "translation (Mueller and Hinnebusch, 1986; Oliveira and\n",
      "McCarthy, 1995).\n",
      "\n",
      "that\n",
      "\n",
      "ACKNOWLEDGMENTS\n",
      "\n",
      "We thank Drs. Athanasios Theologis and Yutaka Ono for pro-\n",
      "viding the rice 17S rRNA cDNA fragment used as probe for the\n",
      "loading control, and Karen Halliday for providing the oligonucle-\n",
      "otides to synthesize the PHYB specific probe. We also thank Drs.\n",
      "Jaime Martinez, Elena Monte, and Debabrata Basu for critical\n",
      "reading of the manuscript.\n",
      "\n",
      "Received April 16, 1999; accepted August 10, 1999.\n",
      "\n",
      "LITERATURE CITED\n",
      "\n",
      "Adam E, Kozma-Bognar L, Dallmann G, Nagy F (1995) Transcrip-\n",
      "tion of tobacco phytochrome-A genes initiates at multiple start\n",
      "sites and requires multiple cis-acting regulatory elements. Plant\n",
      "Mol Biol 29: 983–993\n",
      "\n",
      "Adam E, Szell M, Szekeres J, Schaeffer E, Nagy F (1994) The\n",
      "tobacco\n",
      "\n",
      "and tissue-specific\n",
      "\n",
      "developmental\n",
      "phytochrome-A genes. Plant J 6: 283–293\n",
      "\n",
      "expression of\n",
      "\n",
      "Ayoubi TA, Van De Ven WJ (1996) Regulation of gene expression\n",
      "\n",
      "by alternative promoters. FASEB J 10: 453–460\n",
      "\n",
      "Bruce WB, Deng XW, Quail PH (1991) A negatively acting DNA\n",
      "sequence element mediates phytochrome-directed repression of\n",
      "phyA gene transcription. EMBO J 10: 3015–3024\n",
      "\n",
      "Canto´ n FR, Garcia-Gutierrez A, Gallardo F, Vicente Ad, Canovas\n",
      "FM (1993) Molecular characterization of a cDNA clone encoding\n",
      "glutamine synthetase from a gymnosperm, Pinus sylvestris. Plant\n",
      "Mol Biol 22: 819–828\n",
      "\n",
      "Christensen AH, Quail PH (1989) Structure and expression of a\n",
      "\n",
      "maize phytochrome-encoding gene. Gene 85: 381–390\n",
      "\n",
      "Church GM, Gilbert W (1984) Genomic sequencing. Proc Natl\n",
      "\n",
      "Acad Sci USA 81: 1991–1995\n",
      "\n",
      "Clack T, Mathews S, Sharrock RA (1994) The phytochrome apo-\n",
      "protein family in Arabidopsis is encoded by five genes: the\n",
      "sequences and expression of PHYD and PHYE. Plant Mol Biol\n",
      "25: 413–427\n",
      "\n",
      "Colbert JT, Hershey HP, Quail PH (1983) Autoregulatory control\n",
      "of translatable phytochrome mRNA levels. Proc Natl Acad Sci\n",
      "USA 80: 2248–2252\n",
      "\n",
      "Colbert JT, Hershey HP, Quail PH (1985) Phytochrome regulation\n",
      "\n",
      "of phytochrome mRNA abundance. Plant Mol Biol 5: 91–101\n",
      "\n",
      "Dehesh K, Franci C, Parks BM, Seeley KA, Short TW, Tepper-\n",
      "man JM, Quail PH (1993) Arabidopsis HY8 locus encodes phy-\n",
      "tochrome A. Plant Cell 5: 1081–1088\n",
      "\n",
      "Dehesh K, Franci C, Sharrock RA, Somers DE, Welsch JA, Quail\n",
      "PH (1994) The Arabidopsis phytochrome A gene has multiple\n",
      "transcription start sites and a promoter sequence motif homol-\n",
      "ogous to the repressor element of monocot phytochrome A\n",
      "genes. Photochem Photobiol 59: 379–384\n",
      "\n",
      "Dobres M, Elliot R, Watson J, Thompson W (1987) A phyto-\n",
      "chrome regulated transcript encodes ferredoxin I. Plant Mol Biol\n",
      "8: 53–59\n",
      "\n",
      "Edwards JW, Coruzzi GM (1989) Photorespiration and light act in\n",
      "concert to regulate the expression of the nuclear gene for chlo-\n",
      "roplast glutamine synthetase. Plant Cell 1: 241–248\n",
      "\n",
      "Forreiter C, Cleve Bv, Schmidt A, Apel K (1990) Evidence for a\n",
      "general\n",
      "of NADPH-\n",
      "protochlorophyllide oxidoreductase in angiosperms. Planta 183:\n",
      "126–132\n",
      "\n",
      "light-dependent\n",
      "\n",
      "negative\n",
      "\n",
      "control\n",
      "\n",
      "Furuya M, Ito N, Tomizawa K, Scha¨ffer E (1991) A stable phyto-\n",
      "chrome pool regulates the expression of the phytochrome I gene\n",
      "in pea seedlings. Planta 183: 218–221\n",
      "\n",
      "Gallie DR (1993) Posttranscriptional regulation of gene expression\n",
      "\n",
      "in plants. Annu Rev Plant Physiol Plant Mol Biol 44: 77–105\n",
      "\n",
      "Hershey HP, Barker RF, Idler KB, Lissemore JL, Murray MG,\n",
      "Quail PH (1987) Nucleotide sequence and characterization of a\n",
      "gene encoding the phytochrome polypeptide from Avena. Gene\n",
      "61: 339–348\n",
      "\n",
      "Holzfeind PJ, Ambrose HJ, Newey SE, Nawrotzki RA, Blake DJ,\n",
      "Davies KE (1999) Tissue-selective expression of a-dystrobrevin\n",
      "is determined by multiple promoters. J Biol Chem 274: 6250–\n",
      "6258\n",
      "\n",
      "Johnson E, Bradley M, Harberd NP, Whitelam GC (1994) Photo-\n",
      "responses of light-grown phyA mutants of Arabidopsis. Plant\n",
      "Physiol 105: 141–149\n",
      "\n",
      "Kay SA, Keith B, Shinozaki K, Chye ML, Chua N-H (1989) The\n",
      "rice phytochrome gene: structure, autoregulated expression, and\n",
      "binding of GT-1 to a conserved site in the 59 upstream region.\n",
      "Plant Cell 1: 351–360\n",
      "\n",
      "Kendrick RE, Kronenberg GHM (1994) Photomorphogenesis\n",
      "in Plants, Ed 2. Kluwer Academic Publishers, Dordrecht, The\n",
      "Netherlands\n",
      "\n",
      "Koornneef M, Rolff E, Spruit C (1980) Genetic control of light-\n",
      "thaliana (L.)\n",
      "\n",
      "inhibited hypocotyl elongation in Arabidopsis\n",
      "Heynh. Z Pflanzenphysiol 100: 147–160\n",
      "\n",
      "Lagrange T, Franzetti B, Axelos M, Mache R, Lerbs-Mache S\n",
      "(1993) Structure and expression of the nuclear gene coding for\n",
      "the chloroplast ribosomal protein L21: developmental regulation\n",
      "\n",
      "\fRegulation of PHYA Expression in Arabidopsis\n",
      "\n",
      "1215\n",
      "\n",
      "of a housekeeping gene by alternative promoters. Mol Cell Biol\n",
      "13: 2614–2622\n",
      "\n",
      "Lissemore JL, Colbert JT, Quail PH (1987) Cloning of cDNA for\n",
      "phytochrome from etiolated Cuburbita and coordinate photo-\n",
      "regulation of the abundance of two distinct phytochrome tran-\n",
      "scripts. Plant Mol Biol 8: 485–496\n",
      "\n",
      "Lissemore JL, Quail PH (1988) Rapid transcriptional regulation by\n",
      "phytochromes of the genes for phytochrome and chlorophyll\n",
      "a/b-binding protein in Avena sativa. Mol Cell Biol 8: 4840–4850\n",
      "Mueller PP, Hinnebusch AG (1986) Multiple upstream AUG\n",
      "codons mediate translational control of GCN4. Cell 45: 201–207\n",
      "Myers SJ, Peters J, Huang Y, Comer MB, Barthel F, Dingledine R\n",
      "(1998) Transcriptional regulation of the GluR2 gene: neural-\n",
      "specific expression, multiple promoters, and regulatory ele-\n",
      "ments. J Neurosci 18: 6723–6739\n",
      "\n",
      "Neuhaus G, Bowler C, Hiratsuka K, Yamagata H, Chua N-H\n",
      "(1997) Phytochrome-regulated repression of gene expression re-\n",
      "quires calcium and cGMP. EMBO J 16: 2554–2564\n",
      "\n",
      "Ngai N, Tsai F-Y, Coruzzi G (1997) Light-induced transcriptional\n",
      "repression of the pea AS1 gene: identification of cis-elements\n",
      "and transfactors. Plant J 12: 1021–1034\n",
      "\n",
      "Oliveira CC, McCarthy JEG (1995) The relationship between eu-\n",
      "karyotic translation and mRNA stability. J Biol Chem 270: 8936–\n",
      "8943\n",
      "\n",
      "Quail PH (1991) Phytochrome: a light-activated molecular switch\n",
      "that regulates plant gene expression. Annu Rev Genet 25:\n",
      "389–409\n",
      "\n",
      "Quail PH (1994) Phytochrome genes and their expression. In R\n",
      "Kendrick, G Kronenberg, eds, Photomorphogenesis in Plants, Ed\n",
      "2. Kluwer Academic Publishers, Dordrecht, The Netherlands, pp\n",
      "71–104\n",
      "\n",
      "Quail PH, Boylan MT, Parks BM, Short TW, Xu Y, Wagner D\n",
      "(1995) Phytochromes: photosensory perception and signal trans-\n",
      "duction. Science 268: 675–680\n",
      "\n",
      "Reed JW, Nagatani A, Elich TD, Fagan M, Chory J (1994) Phyto-\n",
      "chrome A and phytochrome B have overlapping but distinct\n",
      "functions in Arabidopsis development. Plant Physiol 104: 1139–\n",
      "1149\n",
      "\n",
      "Reed JW, Nagpal P, Poole DS, Furuya M, Chory J (1993) Muta-\n",
      "tions in the gene for the red/far-red light receptor phytochrome\n",
      "B alter cell elongation and physiological responses throughout\n",
      "Arabidopsis development. Plant Cell 5: 147–157\n",
      "\n",
      "Sato N (1988) Nucleotide sequence and expression of the phyto-\n",
      "chrome gene in Pisum sativum: differential regulation by light of\n",
      "multiple transcripts. Plant Mol Biol 11: 697–710\n",
      "\n",
      "Scha¨fer E, Marchal B, Marme´ D (1972) In vivo measurements of\n",
      "the phytochrome photostationary state in far red light. Photo-\n",
      "chem Photobiol 15: 457–464\n",
      "\n",
      "Schibler U, Sierra F (1987) Alternative promoters in developmen-\n",
      "\n",
      "tal gene expression. Annu Rev Genet 21: 237–257\n",
      "\n",
      "Sharrock RA, Parks BM, Koorneef M, Quail PH (1988) Molecular\n",
      "analysis of the phytochrome deficiency in an Aurea mutant of\n",
      "tomato. Mol Gen Genet 213: 9–14\n",
      "\n",
      "Sharrock RA, Quail PH (1989) Novel phytochrome sequences in\n",
      "Arabidopsis thaliana: structure, evolution, and differential expres-\n",
      "sion of a plant regulatory photoreceptor family. Genes Dev 3:\n",
      "1745–1757\n",
      "\n",
      "Silverthorne J, Tobin EM (1984) Demonstration of transcriptional\n",
      "regulation of specific genes by phytochrome action. Proc Natl\n",
      "Acad Sci USA 81: 1112–1116\n",
      "\n",
      "Silverthorne J, Tobin EM (1987) Phytochrome regulation of nu-\n",
      "\n",
      "clear gene expression. BioEssays 7: 18–23\n",
      "\n",
      "Somers DE, Quail PH (1995) Phytochrome-mediated light regula-\n",
      "tion of PHYA- and PHYB-GUS transgenes in Arabidopsis thaliana\n",
      "seedlings. Plant Physiol 107: 523–534\n",
      "\n",
      "Takaiwa F, Oono K, Sugiura M (1984) The complete nucleotide\n",
      "sequence of a rice 17S rRNA gene. Nucleic Acids Res 12: 5441–\n",
      "5448\n",
      "\n",
      "Terzaghi WB, Cashmore AR (1995) Light-regulated transcription.\n",
      "\n",
      "Annu Rev Plant Physiol Plant Mol Biol 46: 445–474\n",
      "\n",
      "Tobin EM, Kehoe DM (1994) Phytochrome regulated gene expres-\n",
      "\n",
      "sion. Semin Cell Biol 5: 335–346\n",
      "\n",
      "Tomizawa K, Sato N, Furuya M (1989) Phytochrome control of\n",
      "multiple transcripts of the phytochrome gene in Pisum sativum.\n",
      "Plant Mol Biol 12: 295–299\n",
      "\n",
      "Tsai F, Coruzzi G (1990) Dark-induced and organ-specific expres-\n",
      "sion of two asparagine synthetase genes in Pisum sativum. EMBO\n",
      "J 9: 323–332\n",
      "\n",
      "Tunen AJv, Hartman SA, Mur LA, Mol JNM (1989) Regulation of\n",
      "chalcone flavanone isomerase (CHI) gene expression in Petunia\n",
      "hybrida: the use of alternative promoters in corolla, anthers and\n",
      "pollen. Plant Mol Biol 12: 539–551\n",
      "\n",
      "Valvekens D, Montagu MV, Lijsebettens MV (1988) Agrobacte-\n",
      "rium tumefaciens-mediated transformation of Arabidopsis thaliana\n",
      "root explants by using kanamycin selection. Proc Natl Acad Sci\n",
      "USA 85: 5536–5540\n",
      "\n",
      "Whitelam GC, Devlin PF (1997) Roles of different phytochromes\n",
      "in Arabidopsis photomorphogenesis. Plant Cell Environ 20:\n",
      "752–758\n",
      "\n",
      "\f\n"
     ]
    }
   ],
   "source": [
    "id = 'PMC59487'\n",
    "path = r\"C:\\Users\\ammar\\OneDrive\\Desktop\\phoenixbio\\Phoenix Bioinformatics\\PDFs/\"+id+\".pdf\"\n",
    "#path = r\"C:\\Users\\ammar\\OneDrive\\Desktop\\phoenixbio\\Phoenix Bioinformatics\\PDFs\"+id+r\".pdf\"\n",
    "text = convert_pdf_to_txt(path)\n",
    "print(text)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
